### PR TITLE
Add implicit conversion for future option.

### PR DESCRIPTION
**Pull request by fmeriaux**
Add support to retrieve a option of document.

### CODE REVIEW

**Comment by rozza** [comment]
Looks good, needs some tests.

File: driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala [hunk 1]
```diff
@@ -370,6 +370,36 @@ trait ObservableImplicits {
...
+        var subscription: Option[Subscription] = None
+
+        override def onError(throwable: Throwable): Unit = promise.tryFailure(throwable)
+
+        override def onSubscribe(sub: Subscription): Unit = {
+          subscription = Some(sub)
+          sub.request(1)
+        }
+
+        override def onComplete(): Unit = if(!promise.isCompleted) promise.trySuccess(None)
```
Comment (jCalamari) [inline driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala]: There is no point checking for completeness of the promise. `IF THEN` is not atomic.

File: driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala [hunk 2]
```diff
@@ -370,6 +370,36 @@ trait ObservableImplicits {
...
+
+        override def onNext(tResult: T): Unit = {
+          subscription.getOrElse {
```
Comment (jCalamari) [inline driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala]: Violates DRY, copy-paste from `head` method.

File: driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala (hunk 1 above)
Comment (fmeriaux) [inline driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala]: Yes, I fix that.

File: driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala (hunk 2 above)
Comment (fmeriaux) [inline driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala]: I didn't quite understand the problem?

**Comment by rozza** [comment]
Added in 61485c1

**Pull request closed without merging**